site/docs/expressions/field_references.md

```diff
@@ -11,6 +11,59 @@ In Substrait, all fields are dealt with on a positional basis. Field names are o
 | Map KeyExpression         | A wildcard string that is matched against a simplified form of regular expressions. Requires the key type of the map to be a character type. [Format detail needed, intention to include basic regex concepts such as greedy/non-greedy.] | map                | List of map value type     |
 | Masked Complex Expression | An expression that provides a mask over a schema declaring which portions of the schema should be presented. This allows a user to select a portion of a complex object but mask certain subsections of that same object. | any                | any                        |
 
+
+```
+Schema:
+a: struct
+  b: list
+    c: map string
+      x: i32
+
+SQL expression:
+a.b[2].c['my_map_key'].x
+```
+
+```
+Example Protobuf Text:
+
+selection {
+  direct_reference {
+    struct_field {
+      field: 0 # .x
+      child {
+        map_key {
+          map_key {
+            string: "my_map_key" # ['my_map_key']
+          }
+          child {
+            struct_field {
+              field: 0 # .c
+              child {
+                list_element {
+                  offset: 2 # [2]
+                  child {
+                    struct_field {
+                      field: 0 # .b
+                      child {
+                        struct_field {
+                          field: 0 # .a
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  root_reference {
+  }
+}
```

Comment:
I'll repeat my concern from Slack.  I thought, given a reference X.Y.Z that the outer/top-most `struct_field` would be X and the inner/bottom-most `struct_field` would be Z.  However, this example provides the opposite.  The outer-most is Z and the inner-most is X.

If we agree on this definition I will need to update Acero.  I am curious if any other implementations are currently relying on one behavior or another.  Is there a particular reason to prefer this structure?

Reply:
This also feels backwards to me actually. Given `a.b[2].c['my_map_key'].x` I would expect the structure of this to be reversed. That is the:
1. Struct field reference `a`
2. Struct field `b`
3. List offset `2`
4. Struct field `c`
5. Map key `my_map_key`
6. Struct field `x`

I think of this as the series of steps needed to unpack the desired field out of the original input.